File: haskell/multiples_3_5.hs
-- Link to challenge: https://www.codewars.com/kata/514b92a657cdc65150000006/
module MultiplesOf3And5 where

-- Returns the sum of all integers smaller than n that are divisable by 3 or 5
solution :: Integer -> Integer
solution n = sum [x | x <- [1..(n-1)], x `mod` 3 == 0 || x `mod` 5 == 0]

File: c/multiples_3_5.c
// Link to kata: https://www.codewars.com/kata/514b92a657cdc65150000006/

int solution(int number) {
    // Sum all numbers less than number that are multiples of 3 *or* 5
    int sum = 0;

    for (int i = 0; i < number; ++i) {
        if (i % 3 == 0 || i % 5 == 0) {
            sum += i;
        }
    }

    return sum;
}
